### PR TITLE
Allow filename or wildcard for source option

### PR DESCRIPTION
**Pull request by ryuran**
__Breaking change:__ homepage path resolution is now based on working directory if given by commende line argument or on the config file directory.

### CODE REVIEW

File: builder/base/kss_builder_base.js [hunk 1]
```diff
@@ -427,7 +427,7 @@ class KssBuilderBase {
               });
               /* eslint-enable no-loop-func */
             } else if (typeof this.options[key] === 'string') {
-              this.options[key] = path.resolve(this.options[key]);
+              this.options[key] = path.resolve(process.cwd(), this.options[key]);
```
Comment (RobLoach) [inline builder/base/kss_builder_base.js]: Is this `process.cwd()` suppose to be here?

Comment (ryuran) [reply builder/base/kss_builder_base.js]: I think path should be relative to working directory (or config file) no ? where it should be ?

Comment (ryuran) [reply builder/base/kss_builder_base.js]: ping @RobLoach

Comment (RobLoach) [reply builder/base/kss_builder_base.js]: Unsure where it should be based from... process.cwd() seems fine?

File: test/test_kss_builder_base_twig.js [hunk 2]
```diff
@@ -296,7 +296,7 @@ describe('KssBuilderBaseTwig object API', function() {
         source: helperUtils.fixtures('source-twig-builder-test'),
         destination: path.resolve(__dirname, 'output', 'base_twig', 'buildPage'),
         builder: helperUtils.fixtures('builder-twig-with-assets'),
-        homepage: 'alternate-homepage.md'
+        homepage: helperUtils.fixtures('source-twig-builder-test') + '/alternate-homepage.md'
```
Comment (RobLoach) [inline test/test_kss_builder_base_twig.js]: Making use of `path.join()` could save some potential issues on source not being found.

File: builder/base/kss_builder_base.js (hunk 1 above)
Comment (ryuran) [reply builder/base/kss_builder_base.js]: After some test and I need to improve it.
Path in `source` or `homepage` options are not correctly resolved if it's from config file or command line.

Comment (ryuran) [reply builder/base/kss_builder_base.js]: I think it's fixed now

File: test/test_kss_builder_base_twig.js (hunk 2 above)
Comment (ryuran) [reply test/test_kss_builder_base_twig.js]: I see `helperUtils.fixtures` use `path.join` under the hood.

**Comment by ryuran** [comment]
@RobLoach I wait for your validation of it.
You can merge it if you are OK.

**Comment by curtisj44** [comment]
Hi. Are there any plans to release this fix in an upcoming beta?

**Comment by ryuran** [comment]
@curtisj44 There is no plan or I'm not aware of it.

This PR is breaking change and I'm not serene with the release process, @RobLoach can you help ?

**Comment by RobLoach** [comment]
Help with a release? Certainly.... Breaking change means a major release, is there something we need to fix before we hit the button?

**Comment by ryuran** [comment]
> Breaking change means a major release
@RobLoach kss-node is currently in beta but I think every users use the beta because it's tagged as "release" on npm.
I'm not sure about "major" version for breaking change in beta version.

The breaking change is :
> homepage path resolution is now __based on working directory__ if given by command line argument or on the config file directory.

Before, it was __based on source directory__ (`source` option is not a directory path anymore)

**Comment by sntran** [comment]
This change seems to break the path calculation for the Source.

Before, for each elements, the `Source` path was relative to the folder specified in `--source` parameter. After this PR, the `Source` path becomes an absolute full path.

For example:

- Before: `elements.less`
- After: `/Users/sntran/Projects/my-project/css/elements.less`

I have tried to narrow the commits by checking out each ones after the latest release on `npm`. The commit in this PR does indeed break it.

**Comment by RobLoach** [comment]
Would that source change be affected by the following code addition?

``` js
        if (config.source) {
          if (Array.isArray(config.source)) {
            config.source = config.source.map(x => path.resolve(path.dirname(configPath), x));
          } else {
            config.source = path.resolve(path.dirname(configPath), config.source);
          }
        }
```

**Comment by sntran** [comment]
That is very likely. I was just not sure where `configParser` is used.

**Comment by ryuran** [comment]
@sntran Yes, I see.

This PR make `source` option a wildcard and not a directory anymore.
We can’t use a path relative to a wildcard.
It's why path in `Source` in docs are now absolutes.

Maybe we should make it relative to working directory to (cli PWD or config file if used) ?

**Comment by sntran** [comment]
Absolute path in `Source` is not preferable. The generated HTML can be moved to different file system, different server (i.e. Github Pages). We don't want to expose the full file system path of the user generated the HTML. Even so, that `Source` path is now meaningless when the styleguide is moved.

Maybe we can update the base builder to convert the absolute path to relative path?

**Comment by ryuran** [comment]
@sntran I fix it in #549 I hope for your review 😉

**Pull request merged**